### PR TITLE
Add application credentials description placeholder 

### PR DESCRIPTION
**Pull request by allenporter**
## Proposed change

Add an application credentials description placeholder for integration specific customizations.

Screen Shot 2022-06-04 at 12 26 33 AM

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request: https://github.com/home-assistant/developers.home-assistant/pull/1356
- Link to core PR: https://github.com/home-assistant/core/pull/73014

## Checklist

- [X] The code change is tested and works locally.
- [X] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by MartinHjelmare** [comment]
Core PR is approved.

File: src/panels/config/application_credentials/dialog-add-application-credential.ts [hunk 1]
```diff
@@ -61,11 +69,15 @@ export class DialogAddApplicationCredential extends LitElement {
...
-    this._domains = config.domains.map((domain) => ({
-      id: domain,
-      name: domainToName(this.hass.localize, domain),
-    }));
+    this._config = await fetchApplicationCredentialsConfig(this.hass);
+    this._domains = Object.entries(this._config.integrations).map(
+      ([domain, info]) => ({
+        id: domain,
+        name: domainToName(this.hass.localize, domain),
+        description_placeholders: info.description_placeholders,
```
Comment (bramkragten) [inline src/panels/config/application_credentials/dialog-add-application-credential.ts:77]: This is never used? We always use `this._config` directly when fetching `description_placeholders`?

**Review by bramkragten — DISMISSED** [review]

File: src/panels/config/application_credentials/dialog-add-application-credential.ts (hunk 1 above)
Comment (allenporter) [reply src/panels/config/application_credentials/dialog-add-application-credential.ts]: Yes, good catch, this was my first attempt.  Removed!

**Review by bramkragten — APPROVED** [review]

**Comment by allenporter** [comment]
Core PR was merged.

**Pull request merged**